scripts/west_commands/runners/intel_adsp.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Runner for flashing with the Intel ADSP CAVS boards.'''
+
+import os
+import sys
+import re
+import hashlib
+import random
+import shutil
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+from zephyr_ext_common import ZEPHYR_BASE
+
+DEFAULT_CAVSTOOL='soc/xtensa/intel_adsp/tools/cavstool_client.py'
+DEFAULT_SOF_MOD_DIR=os.path.join(ZEPHYR_BASE, '../modules/audio/sof')
+DEFAULT_RIMAGE_TOOL=shutil.which('rimage')
+DEFAULT_CONFIG_DIR=os.path.join(DEFAULT_SOF_MOD_DIR, 'rimage/config')
+DEFAULT_KEY_DIR=os.path.join(DEFAULT_SOF_MOD_DIR, 'keys')
+
+
+class IntelAdspBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for the intel ADSP boards.'''
+
+    def __init__(self,
+                 cfg,
+                 remote_host,
+                 rimage_tool,
+                 config_dir,
+                 default_key,
+                 key,
+                 run_twister
+                 ):
+        super().__init__(cfg)
+
+        self.remote_host = remote_host
+        self.rimage_tool = rimage_tool
+        self.config_dir = config_dir
+        self.bin_fw = os.path.join(cfg.build_dir, 'zephyr', 'zephyr.ri')
+        self.bin_file = os.path.join(cfg.build_dir, 'zephyr', 'zephyr.bin')
```

Comment:
I think you can just use `cfg.bin_file` instead of reconstructing the path

Reply:
Thanks for correcting this! And also I found it doesn't need bin_file now, so remove it directly. If there is in a case needing bin_fw again, I will use the cfg.bin_file directly, thanks.